R/plotPValHist.R

```diff
@@ -25,110 +24,146 @@
 #' }
 #'
 #' @import ggplot2
-#' @importFrom dplyr filter
+#' @importFrom dplyr filter select
 #' @importFrom canvasXpress canvasXpress
 #'
 #' @export
 plotPvalHist <- function(P.Val,
                          plotType = "canvasXpress",
-                         facet = TRUE,
-                         binWidth = 0.02,
-                         alpha = 0.6,
-                         color = "dodgerblue3",
-                         fill = "dodgerblue3") {
+                         facet          = TRUE,
+                         binWidth       = 0.02,
+                         transparency   = 0.6,
+                         color    = "dodgerblue3") {
 
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
+    plotType = tolower(plotType)
+
+    assertthat::assert_that(!missing(P.Val),
+                            class(P.Val)[[1]] %in% c("matrix","data.frame"),
+                            msg = "P.Val must be specified and must be of class matrix or dataframe.")
+
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
 
+    if (!assertthat::see_if(is.character(color),
```

Comment:
Why **!** is at the beginning of the statement and not in the condition itself **length != 1**

Reply:
Mohammed, see_if returns a False if any of the criteria is not satisfied. So I negate it to ensure it goes into the loop.